text/0000-improved-ponybench.md

```diff
@@ -0,0 +1,175 @@
+- Feature Name: improved-ponybench
+- Start Date: 2018-01-15
+- RFC PR:
+- Pony Issue:
+
+# Summary
+
+The contents of this RFC will present concerns with the current implementation of ponybench and provide an upgrade path based on a working prototype at https://github.com/Theodus/pony-benchmark.
+
+# Motivation
+
+Ponybench, in its current form, has flaws that make it unusable in some cases:
+- Benchmark functions cannot be garbage collected.
+- The user has no way to specify actions that should occur for setup or tear-down.
+- Asynchronous benchmarks have a tendency to move to other threads due to the use of Promises, leading to high deviation in the results.
+
+The new version of ponybench will address these concerns. New functionality will also be added to produce statistics from samples and to output the sample data in CSV format so that users may further analyze the data collected.
```

Comment:
`The new version of ponybench will address these concerns` seems to imply this new version of ponybench will ensure that the benchmark runner actor will not be moved to another scheduler thread.

How are you ensuring this?

Reply:
I don't know of any way to ensure that the user will not have other actors that will be scheduled during a benchmark, or that the runtime won't move the actor. However, async benchmarks no longer rely on promises (or any other actor) to run the benchmark. Though the user will likely have their own actors being tested in this case.

Reply:
Makes sense. Thanks for the clarification.